.github/workflows/docs.yml

```diff
@@ -0,0 +1,85 @@
+name: docs
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - .pre-commit-config.yaml
+      - .github/workflows/docs.yml
+      - '**.py'
+      - '**.ipynb'
+      - '**.html'
+      - '**.jinja'
+      - '**.js'
+      - poetry.lock
+      - pyproject.toml
+      - '**.rst'
+      - '**.md'
+      - docs/cyclops-webpage
+    types: [opened, synchronize, closed]
+  push:
+    branches:
+      - main
+    paths:
+      - .pre-commit-config.yaml
+      - .github/workflows/docs.yml
+      - '**.py'
+      - '**.ipynb'
+      - '**.html'
+      - '**.jinja'
+      - '**.js'
+      - poetry.lock
+      - pyproject.toml
+      - '**.rst'
+      - '**.md'
+      - docs/cyclops-webpage
+
+jobs:
+  build:
+    if: github.event_name == 'push' || (github.event_name == 'pull_request' && (github.event.action == 'opened' || github.event.action == 'synchronize' || (github.event.action == 'closed' && github.event.pull_request.merged == true)))
```

Comment:
Wouldn't a PR being closed and merged trigger a 'push' event on the main branch?
```
(github.event.action == 'closed' && github.event.pull_request.merged == true) == (github.event_name == 'push')
```

Reply:
Yeah, i want this to run again when the PR is merged. So once when the PR is opened, then every time it is updated. Then finally when it is merged.

Reply:
Doesn't that mean it runs twice when the PR is merged? Once on the main branch because of the 'push' event and again on the branch with the PR that just got merged.

Reply:
Ohh maybe the confusion is that i also want it to run when there is just a normal push event to main. I have disabled direct pushes to the main branch but it can be overridden by admins. So in that cases it would still run the workflow.

Reply:
I understand that. I just meant that closing and merging a PR would trigger a push event on main, so there may not be a need for `(github.event.action == 'closed' && github.event.pull_request.merged == true)` because `github.event_name == 'push'` will be true for the main branch.

I see that's not the case with the recent merge.